📖 update releasing.md with release branches (#1298)
metal3-io/baremetal-operator: CONTRIBUTING.md

#### Backporting

We only accept backports of critical bugs, security issues, or bugs without
easy workarounds, any
backport MUST not be breaking for either API or behavioral changes.
We generally do not accept PRs against older release branches.

## Branches

> Baremetal Operator has only the *main* branch currently.

Reviewer: This should be updated I think.
```suggestion
Baremetal Operator has the *main* branch and *release-X* branches.
```

Author: Fixed.
